csrc/id_model/validation_utils.h

```diff
@@ -0,0 +1,22 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+
+#include <compute_at_map.h>
+#include <id_model/id_model.h>
+#include <val_graph.h>
+
+namespace nvfuser {
+
+class IdModelValidator {
```

Comment:
Not sure we need a class. A class with a static method can just be a method maybe under a certain namespace?

Reply:
Yeah, I don't know why I put it in a class. I'll just remove the class.

Reply:
Actually, I remembered why I chose to have a class. This validation uses `ComputeAtMap` in an unintended way and needs to access private members, so I worked around it by making it a friend. Since I expect there would be similar validation functions but for different types of graphs, I thought it would be convenient to have a single friend class rather than listing each function as a friend.

I'll keep this as is. We could of course change `ComputeAtMap` so that we wouldn't need to use friend, but this is just a temporary class to make the transition to the new infrastructure go smoothly, so I chose the way that works quickly.

Reply:
Good to know!